send_sms.py

```diff
@@ -0,0 +1,29 @@
+import requests
+import json
+import general
+from config import URL, APIKEY, SECRETKEY, PHONENO
+
+# sending post request
+def send_post_request(reqUrl, apiKey, secretKey, useType, phoneNo, senderId, textMessage):
+  req_params = {
+  'apikey':apiKey,
+  'secret':secretKey,
+  'usetype':useType,
+  'phone': phoneNo,
+  'message':textMessage,
+  'senderid':senderId
+  }
+  return requests.post(reqUrl, req_params)
+
+
+def send_sms(message):
+    try:
+        # getting response and save as response object
+        response = send_post_request(URL, APIKEY, SECRETKEY, 'stage', PHONENO, 'WAYSMS', message )
+        # print response if you want
+        print (response.text)
+    except Exception as e:
+        print(e)
+    finally :
+        print()
+        exit()
```

Comment:
We might not want to exit if we couldn't send the SMS. For instance, I was running this script on an AWS instance 24x7. If the SMS failed, it would've stopped the program. This is what we would want.

Reply:
Hey @jarifibrahim if the SMS failed, then it would have stopped the program or not?

Reply:
At this moment, I can't understand what to do if SMS failed. Can you help me?
If SMS failed then I would need to have exit from program or  have to continue?

Reply:
No. We should not kill the program if the sms failed. Let's keep the programming running even if the sms failed to send.

Reply:
Hey @jarifibrahim I did following changes. Can you review this?

```
def send_sms(message):
    try:
        # get response
        response = sendPostRequest(WAY2SMS_URL, SMS_API_KEY, SMS_SECRET_KEY, 'stage', PHONE_NO, 'WAYSMS', message )
        # print response if you want
        res =  response.text
        if 'success' in res:
            print('\033[92m' + 'Successfully sent the sms' + '\033[0m')
    except requests.exceptions.HTTPError as eh:
        print('\033[91m' + 'An HTTP error occurred. %s', %eh)
        print('\033[0m') 
    except requests.exceptions.ConnectionError as ec:
        print('\033[91m' + 'A Connection error occurred %s', %ec)
        print('\033[0m')
    except requests.exceptions.Timeout as et:
        print('\033[91m' + 'It is ConnectTimeout or ReadTimeout %s', %errt)
        print('\033[0m')
    except Exception as e:
        print('\033[91m' + "Something went wrong. Failed to send SMS: %s" % e)
        print('\033[0m')
    finally :
         pass
```

Reply:
@nishikeshKardak I think we don't need to handle all the exceptions. Look at this example
```python
>>> def try_and_print(req, params):
...  try:
...   requests.post(req, params)
...  except Exception as e:
...   print("Failed to send SMS: %s" % e)
>>> try_and_print("foo", "bar")
Failed to send SMS: Invalid URL 'foo': No schema supplied. Perhaps you meant http://foo?
>>> try_and_print("http:://google.com", "bar")
Failed to send SMS: No connection adapters were found for 'http:://google.com/'
>>> try_and_print("http://google.com", "bar")
>>> try_and_print("http://localhost", "bar")
Failed to send SMS: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f89424932d0>: Failed to establish a new connection: [Errno 111] Connection refused',))
```

So a simple
```python
except Exception as e:
    print("Failed to send SMS: %s " % e)
```
should be enough